### PR TITLE
ENS api

### PR DESCRIPTION
**Pull request by Sprimage**
Added some of Conor's React to ENS api i wrote. Let me know if this is also in the right direction please

### CODE REVIEW

File: src/components/Address/Address.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
+import React, { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import ENS from 'ethereum-ens'
+import { isValid } from 'ipaddr.js'
```
Comment (ConorOkus) [inline src/components/Address/Address.js]: Think this has accidentally been imported?

File: src/components/Address/Address.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+          })
+        } else {
+          resolveName(debouncedSearchTerm, ens).then(result => {
+            setIsResolving(false)
+            setResolvedAddress(result)
+          })
+        }
+      } else {
+        setIsResolving(false)
+        alert('wrong input')
```
Comment (ConorOkus) [inline src/components/Address/Address.js]: Maybe set some state for an error message here?

File: src/components/Address/Address.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    .reverse(search)
+    .addr()
+    .then(addr => addr)
+    .catch(err => {
+      console.error(err)
+      return 'Unable to find address'
+    })
+}
+
+const AddressInput = ({ provider, className }) => {
```
Comment (ConorOkus) [inline src/components/Address/Address.js]: Remember to add className to prop types and pass it to the rendered component

File: src/components/Address/Address.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+          resolveAddress(debouncedSearchTerm, ens).then(result => {
+            setIsResolving(false)
+
+            setResolvedAddress(result)
+          })
+        } else {
+          resolveName(debouncedSearchTerm, ens).then(result => {
+            setIsResolving(false)
+            setResolvedAddress(result)
+          })
```
Comment (ConorOkus) [inline src/components/Address/Address.js]: Very simple, I like it 👍

File: src/components/Address/Address.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    } else {
+      setIsResolving(false)
+    }
+  }, [debouncedSearchTerm])
+
+  return (
+    <>
+      <label>Search ENS name </label>
+      <input
+        onChange={async e => {
```
Comment (ConorOkus) [inline src/components/Address/Address.js]: Can remove the async from here

**Comment by Sprimage** [comment]
@ConorOkus made some new commits. Regarding the styling, will it use inline style or from external css? @Arachnid Maybe you could chip in and provide some insight? Especially regarding the consumer having access to the resolved address

**Comment by jefflau** [comment]
Thank you for the contribution. We have deployed a version of the Address Component from @industral. Closing this PR.

**Pull request closed without merging**